src/libraries/System.Private.CoreLib/src/System/DateOnly.cs

```diff
@@ -0,0 +1,904 @@
+        /// <returns>A string representation of value of the current DateOnly object as specified by format and provider.</returns>
+        public string ToString(string? format, IFormatProvider? provider)
+        {
+            if (format == null || format.Length == 0)
+            {
+                format = "d";
+            }
+
+            if (format.Length == 1)
+            {
+                switch (format[0])
+                {
+                    case 'o':
+                    case 'O':
+                        {
+                            Span<char> destination = stackalloc char[10];
+                            bool b = DateTimeFormat.TryFormatDateOnlyO(Year, Month, Day, destination);
+                            Debug.Assert(b);
+
+                            return destination.ToString();
+                        }
+
+                    case 'r':
+                    case 'R':
+                        {
+                            Span<char> destination = stackalloc char[16];
+                            bool b = DateTimeFormat.TryFormatDateOnlyR(DayOfWeek, Year, Month, Day, destination);
+                            Debug.Assert(b);
+
+                            return destination.ToString();
+                        }
+
+                    case 'm':
+                    case 'M':
+                    case 'd':
+                    case 'D':
+                    case 'y':
+                    case 'Y':
+                        return DateTimeFormat.Format(GetEquivalentDateTime(), format, provider);
+
+                    default:
+                        throw new FormatException(SR.Format_InvalidString);
+                }
+            }
+
+            DateTimeFormat.IsValidCustomDateFormat(format.AsSpan(), throwOnError: true);
+            return DateTimeFormat.Format(GetEquivalentDateTime(), format, provider);
+        }
+
+        /// <summary>
+        /// Tries to format the value of the current DateOnly instance into the provided span of characters.
+        /// </summary>
+        /// <param name="destination">When this method returns, this instance's value formatted as a span of characters.</param>
+        /// <param name="charsWritten">When this method returns, the number of characters that were written in destination.</param>
+        /// <param name="format">A span containing the characters that represent a standard or custom format string that defines the acceptable format for destination.</param>
+        /// <param name="provider">An optional object that supplies culture-specific formatting information for destination.</param>
+        /// <returns>true if the formatting was successful; otherwise, false.</returns>
+        public bool TryFormat(Span<char> destination, out int charsWritten, ReadOnlySpan<char> format = default(ReadOnlySpan<char>), IFormatProvider? provider = null)
+        {
+            if (format.Length == 0)
+            {
+                format = "d".AsSpan();
```

Comment:
Nit: the `.AsSpan()` isn't needed
```suggestion
                format = "d";
```

Reply:
will do